draft-vvv-webtransport-http3.md

```diff
@@ -325,12 +310,11 @@ variable-length integer, followed by the user-specified stream data
 
 ## Datagrams
 
-Datagrams can be sent using the DATAGRAM frame as defined in
-[QUIC-DATAGRAM] and {{!H3-DATAGRAM=I-D.schinazi-quic-h3-datagram}}.  For all
-HTTP/3 connections in which the `http3_transport_support` option is negotiated,
-the Flow Identifier is set to the session ID.  In other words, the format of
-datagrams SHALL be the session ID, followed by the user-specified payload
-({{fig-datagram}}).
+Datagrams can be sent using the DATAGRAM frame as defined in [QUIC-DATAGRAM]
+and [HTTP3-DATAGRAM].  For all HTTP/3 connections in which the
+SETTINGS_ENABLE_WEBTRANSPORT option is negotiated, the Flow Identifier is set
+to the session ID.  In other words, the format of datagrams SHALL be the
```

Comment:
This goes against the design of draft-schinazi-quic-h3-datagram. Should we use the same mechanism as CONNECT-UDP to allow multiplexing CONNECT-UDP and WebTransport over the same h3 connection?

Reply:
We should!  Wanna update CONNECT-UDP draft to use stream IDs?

Reply:
No, CONNECT-UDP relies on having separate namespaces for streams and DATAGRAM flow IDs. This feature is used by [the quic-proxy extension](https://tools.ietf.org/html/draft-pauly-masque-quic-proxy-00)

Reply:
I think it would be ok to have the same flow ID vendor work independent of the semantic usage of the connected stream right?

Reply:
I am not convinced CONNECT-UDP actually relies on this property, at least the draft you link shows no indication on why that is required.

Reply:
@vasilvv I think you're asking a similar question to the one I posed on https://github.com/ietf-wg-masque/draft-ietf-masque-connect-udp/issues/16

Reply:
@vasilvv if you look at the [Example section](https://tools.ietf.org/html/draft-pauly-masque-quic-proxy-00#section-6) you'll see that multiple CONNECT-UDP requests with this extension can share a flow ID

Reply:
I read that, I just don't see why they need to share it.

Reply:
It's about shared lifetimes - it allows multiple requests to come and go and the underlying flow ID to remain valid - otherwise as the older flow IDs get removed you could end up with dropped datagrams. This also allows using smaller encodings for flow ID varints which adds up over time

Reply:
I'm confused.  You have to keep the stream up for the duration of associated connection ID's lifetime anyways, since once the stream is gone, so are the mappings.

Does the varint thing matter?  I thought the entire point of draft-pauly-masque-quic-proxy is to bring 99% of the traffic outside the DATAGRAMs.

Reply:
You keep the stream open for its part of the mapping, but you can think of the flow ID as being reference counted by all the streams that use it.

Conceptually, HTTP/3 DATAGRAM Flow IDs have their own namespace similar to how HTTP/3 Push IDs have their own namespace distinct from the Stream ID namespace.

Reply:
This all sounds like unnecessary complexity to me.  If you don't keep separate flow IDs, you don't need to reference-count them in first place.

Reply:
Let's talk about this on VC sometime, since I'd like to walk through all the pros and cons here.

Reply:
Let's merge this as-is for now, and update when with have consensus in the h3datagram draft.